sleap_io/io/nwb.py

```diff
@@ -14,21 +14,131 @@
+        )
+    return Labels(labeled_frames)
+
+
+def labels_to_pose_training(labels: Labels, filename: str, **kwargs) -> PoseTraining:  # type: ignore[return]
+    """Creates an NWB PoseTraining object from a Labels object."""
+    training_frame_list = []
+    for i, labeled_frame in enumerate(labels.labeled_frames):
+        training_frame_name = name_generator("training_frame")
+        training_frame_annotator = f"{training_frame_name}{i}"
+        training_frame_skeleton_instances = SkeletonInstances(
+            [
+                instance_to_skeleton_instance(instance)
+                for instance in labeled_frame.instances
+            ]
+        )
+        training_frame_video = labeled_frame.video
+        training_frame_video_index = labeled_frame.frame_idx
+        training_frame = TrainingFrame(
+            name=training_frame_name,
+            annotator=training_frame_annotator,
+            skeleton_instances=training_frame_skeleton_instances,
+            source_video=training_frame_video.filename,
+            source_video_frame_index=training_frame_video_index,
+        )
+        training_frame_list.append(training_frame)
+    training_frames = TrainingFrames(training_frames=training_frame_list)
+
+    source_videos = labels.videos[0].filename
+    pose_training = PoseTraining(
+        training_frames=training_frames, source_videos=source_videos
+    )
+    return pose_training
+
+def slp_skeleton_to_nwb(skeleton: SLEAPSkeleton) -> NWBSkeleton: # type: ignore[return]
+    """Converts SLEAP skeleton to NWB skeleton."""
+    nwb_edges: list[list[int, int]] = []
+    for i, _ in enumerate(skeleton.edges):
+        if i == len(skeleton.edges):
+            break
+        nwb_edges.append([i, i + 1])
+    return NWBSkeleton(
+        name=skeleton.name,
+        nodes=skeleton.node_names,
+        edges=np.array(nwb_edges, dtype=np.uint8),
+    )
+
+
+def instance_to_skeleton_instance(instance: Instance) -> SkeletonInstance: # type: ignore[return]
+    skeleton = slp_skeleton_to_nwb(instance.skeleton)
+    return SkeletonInstance(
+        id=np.uint(10),
+        node_locations=skeleton.edges,
+        node_visibility=[True, False],
+        skeleton=skeleton,
+    )
+
+
+def videos_to_source_videos(videos: List[Video]) -> SourceVideos: # type: ignore[return]
+    """Converts a list of SLEAP Videos to NWB SourceVideos."""
+    source_videos = []
+    for video in videos:
+        image_series = ImageSeries(
+            name=video.filename,
+            description="Video file",
+            unit="NA",
+            format="external",
+            external_file=[video.filename],
+            dimension=[video.backend.height, video.backend.width],
+            starting_frame=[0],
+            rate=30.0,
+        )
+        source_videos.append(image_series)
+    return SourceVideos(data=source_videos)
+
+
 def get_timestamps(series: PoseEstimationSeries) -> np.ndarray:
```

Comment:
`PoseEstimationSeries` was already complete, so you shouldn't need too many modification to that part of the code